### ISSUE TITLE
Chrome may override disabled autoComplete in TextFields

### ISSUE THREAD

**Issue by marina101** [Shopify/polaris-react#1244]
# Issue summary

Chrome is known to sometimes ignore the autoComplete=false setting on inputs: https://bugs.chromium.org/p/chromium/issues/detail?id=370363#c7

The workaround is to set it to a string field named differently from the ones autocomplete looks for:
https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill

## Expected behavior

TextField should be able to handle a string prop as well as a boolean prop for autocomplete, and then set autocomplete to the value of the string if the prop is a string.

## Actual behavior

TextField only accepts boolean or null values.

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: 3.10.0
- Browser: Chrome Version 72.0.3626.121 (Official Build) (64-bit)
- Device: mac
- Operating System: Mojave 10.14.3

**Comment by marina101** [comment]
After discussion, it seems that new browsers might only be respecting it if it set to false: 
https://github.com/Shopify/polaris-react/pull/708
https://github.com/Shopify/polaris-react/pull/761

**Comment by marina101** [comment]
I'm re-opening this. We're getting multiple shops for whom chrome is ignoring the autocomplete=off setting. I was able to reproduce this behavior in production using chrome 72, and when I changed the value of autocomplete to a string we got the correct behavior. I'd like to add the option of passing in a string argument instead of a boolean to TextField.

Longer term this may be related to https://developers.google.com/web/updates/2015/06/checkout-faster-with-autofill where chrome treats autocomplete and autofill differently and we might need to add support for autofill explicitly, but in the short term, passing in a string for autocomplete can fix this situation for us, which we'd like to do soon as there are merchants depending on this being fixed to be able to get their payments.
